pkg/audit/manager.go

```diff
@@ -274,62 +275,107 @@ func (am *Manager) auditResources(
 	var errs opa.Errors
 	nsCache := newNSCache()
 
-	for gv, gvKinds := range clusterAPIResources {
-		for kind := range gvKinds {
-			objList := &unstructured.UnstructuredList{}
-			opts := &client.ListOptions{
-				Limit: int64(*auditChunkSize),
+	constraintsList := make(map[string]bool)
+	if *auditMatchKindOnly {
+		constraintList := &unstructured.UnstructuredList{}
+		for _, c := range resourceList {
+			constraintList.SetGroupVersionKind(c)
+			if err = am.client.List(ctx, constraintList); err != nil {
+				am.log.Error(err, "Unable to list objects for gvk", "group", c.Group, "version", c.Version, "kind", c.Kind)
+				continue
 			}
-			resourceVersion := ""
-
-			for {
-				objList.SetGroupVersionKind(schema.GroupVersionKind{
-					Group:   gv.Group,
-					Version: gv.Version,
-					Kind:    kind + "List",
-				})
-				objList.SetResourceVersion(resourceVersion)
-
-				err := am.client.List(ctx, objList, opts)
+			for _, constraint := range constraintList.Items {
+				kinds, found, err := unstructured.NestedSlice(constraint.Object, "spec", "match", "kinds")
 				if err != nil {
-					am.log.Error(err, "Unable to list objects for gvk", "group", gv.Group, "version", gv.Version, "kind", kind)
+					am.log.Error(err, "Unable to return spec.match.kinds field", "group", c.Group, "version", c.Version, "kind", c.Kind)
```

Comment:
Errors here should probably default to using "*"

Reply:
done